py/usageaccountant/datadog_fetcher.py

```diff
@@ -0,0 +1,292 @@
+    pointlist: List[List[float]]
+    scope: str
+
+
+class DatadogResponse(DatadogResponseSeries):
+    error: Optional[str]
+    query: str
+    from_date: int
+    to_date: int
+    status: str
+    series: List[DatadogResponseSeries]
+
+
+def is_valid_query(query: str) -> bool:
+    """
+    Validates if query string contains shared_resource_id
+    and app_feature–parameters used in UsageAccountant.record().
+
+    query: Datadog query
+    """
+    return "shared_resource_id" in query and "app_feature" in query
+
+
+def get(query: str, start_time: int, end_time: int) -> Any:
+    """
+    Fetches timeseries data from Datadog API, returns response dict.
+
+    query: Datadog query
+    start_time: Start of the time window in Unix epoch format
+                with second-level precision
+    end_time: End of the time window in Unix epoch format
+              with second-level precision
+    """
+    data = {"query": query, "from": start_time, "to": end_time}
+
+    base_url = "https://api.datadoghq.com/api/v1/query"
+    query_string = urllib.parse.urlencode(data)
+    full_url = f"{base_url}?{query_string}"
+
+    try:
+        request = Request(url=full_url, headers=headers)
+        # context is needed to handle
+        # ssl.SSLCertVerificationError:
+        # [SSL: CERTIFICATE_VERIFY_FAILED] on macOS
+        # TODO remove next three lines and
+        # context parameter from the call to urlopen()
+        import ssl
+
+        context = ssl.SSLContext()
+        response = urlopen(request, context=context).read()
+        return json.loads(response)
+
+    except Exception as e:
+        raise e
+
+
+def parse_response_series_and_unit(
+    response: DatadogResponse,
+) -> Tuple[List[DatadogResponseSeries], UsageUnit]:
+    """
+    Extracts list of series objects and unit from the API response.
+
+    response: response received from API call
+    """
+
+    error_msg = response.get("error")
+    if error_msg:
+        raise Exception(format_exception_message(response, error_msg))
+
+    if response.get("series") is not None:
+        series_list = response.get("series", [])
+        if len(series_list) > 0:
+            first_series = series_list[0]
+            if first_series.get("unit") is not None:
+                unit_list = first_series.get("unit")
+                if isinstance(unit_list, List) and len(unit_list) > 0:
+                    first_unit = unit_list[0]
+                    if first_unit.get("plural"):
+                        # assumes all series have one and the same unit
+                        response_unit = first_unit.get("plural")
+
+    if len(series_list) == 0:
+        msg = "No timeseries data found in response."
+        raise Exception(format_exception_message(response, msg))
+
+    if response_unit is None:
+        msg = "No unit found in response."
+        raise Exception(format_exception_message(response, msg))
+
+    try:
+        parsed_unit = UsageUnit(response_unit.lower())
```

Comment:
I am not sure you can assume the unit you will get from datadog will map one to one to the one you need to pass to the UsageUnit. The unit in datadog depends on how the metrics is recorded, it may not be correct or the unit contained in the metrics could be a multiple of the one the accountant expect.
You can try a few times but please allow the user to specify the unit together with the query to run.

Reply:
It is likely that `UsageUnit` does not exhaustively list all the units.
I was concerned about `UsageAccountant` breaking, specifically this [line](https://github.com/getsentry/usage-accountant/blob/aca9b26f18409f43536c676cb706c94cdc227546/py/usageaccountant/accumulator.py#L152), in case the unit passed is not in the Enum. 
Is it a concern valid?

Reply:
`UsageUnit` is not going to be flexible. That is by design, we support only certain units, so we know what we can compare with what.
Yes the concern is valid, please consider my suggestion in the comment above and ask the user to configure the unit in their config file rather than trying to infer it from the datadog api call.